fast64_internal/oot/oot_cutscene.py

```diff
@@ -0,0 +1,604 @@
+
+
+################################################################################
+# Operators and panel
+################################################################################
+
+def checkGetFilePaths(context):
+	cpath = bpy.path.abspath(context.scene.ootCutsceneExportPath)
+	if not cpath.endswith('.c'):
+		raise PluginError('Output file must end with .c')
+	hpath = cpath[:-1] + 'h'
+	headerfilename = hpath.split('/')[-1]
+	return cpath, hpath, headerfilename
+
+class OOT_ExportCutscene(bpy.types.Operator):
+	# set bl_ properties
+	bl_idname = 'object.oot_export_cutscene'
+	bl_label = "Export Cutscene"
+	bl_options = {'REGISTER', 'UNDO', 'PRESET'}
+	
+	def execute(self, context):
+		try:
+			if context.mode != 'OBJECT':
+				bpy.ops.object.mode_set(mode = "OBJECT")
+			activeObj = context.view_layer.objects.active
+			if activeObj is None or activeObj.data is not None or activeObj.ootEmptyType != 'Cutscene':
+				raise PluginError('You must select a cutscene object')
+			if activeObj.parent is not None:
+				raise PluginError('Cutscene object must not be parented to anything')
+			cpath, hpath, headerfilename = checkGetFilePaths(context)
+			csdata = ootCutsceneIncludes(headerfilename)
+			converted = convertCutsceneObject(activeObj)
+			csdata.append(ootCutsceneDataToC(converted, converted.name))
+			writeCData(csdata, hpath, cpath)
+			self.report({'INFO'}, 'Successfully exported cutscene')
+			return {'FINISHED'}
+		except Exception as e:
+			raisePluginError(self, e)
+			return {'CANCELLED'} # must return a set
+
+class OOT_ExportAllCutscenes(bpy.types.Operator):
+	# set bl_ properties
+	bl_idname = 'object.oot_export_all_cutscenes'
+	bl_label = "Export All Cutscenes"
+	bl_options = {'REGISTER', 'UNDO', 'PRESET'}
+	
+	def execute(self, context):
+		try:
+			if context.mode != 'OBJECT':
+				bpy.ops.object.mode_set(mode = "OBJECT")
+			cpath, hpath, headerfilename = checkGetFilePaths(context)
+			csdata = ootCutsceneIncludes(headerfilename)
+			count = 0
+			for obj in context.view_layer.objects:
+				if obj.data is not None or obj.ootEmptyType != 'Cutscene':
+					continue
+				if obj.parent is not None:
+					raise PluginError('Cutscene object must not be parented to anything')
+				converted = convertCutsceneObject(obj)
+				csdata.append(ootCutsceneDataToC(converted, converted.name))
+				count += 1
+			if count == 0:
+				raise PluginError('Could not find any cutscenes to export')
+			writeCData(csdata, hpath, cpath)
+			self.report({'INFO'}, 'Successfully exported ' + str(count) + ' cutscenes')
+			return {'FINISHED'}
+		except Exception as e:
+			raisePluginError(self, e)
+			return {'CANCELLED'} # must return a set
+
+class OOT_ExportCutscenePanel(bpy.types.Panel):
+	bl_idname = "OOT_PT_export_cutscene"
+	bl_label = "OOT Cutscene Exporter"
+	bl_space_type = 'VIEW_3D'
+	bl_region_type = 'UI'
+	bl_category = 'OOT'
+	
+	@classmethod
+	def poll(cls, context):
+		return True
+
+	# called every frame
+	def draw(self, context):
```

Comment:
```suggestion
	def draw(self, context):
```
that comment is wrong

Reply:
You are correct, it is called when there are mouse movements over the panel area. I think I copied that from some Panel tutorial. (This is code which was already in fast64, just moved to this file, by the way.)